src/plugins/intel_gna/frontend/layer_quantizer.hpp

```diff
@@ -4,99 +4,84 @@
 namespace frontend {
 
-/**
- * @brief description of quantisation precision
- * @tparam Ip - input precision
- * @tparam Op - output precision
- * @tparam Wp - weights precision
- * @tparam Bp - biases precision
- * @tparam Np - network precision - can be auto generated in future
- */
-template <class Ip, class Op, class Wp, class Bp, class Np>
-struct QuantDescTmpl {
-    using WeightsPrecision = Wp;
-    using BiasesPrecision = Bp;
-
-    InferenceEngine::TPrecision<Ip> _Ip;
-    InferenceEngine::TPrecision<Op> _Op;
-    InferenceEngine::TPrecision<Wp> _Wp;
-    InferenceEngine::TPrecision<Bp> _Bp;
-
-    QuantDescTmpl() = default;
-    QuantDescTmpl(InferenceEngine::TPrecision<Ip> _Ip,
-              InferenceEngine::TPrecision<Op> _Op,
-              InferenceEngine::TPrecision<Wp> _Wp,
-              InferenceEngine::TPrecision<Bp> _Bp) : _Op(_Op), _Ip(_Ip), _Wp(_Wp), _Bp(_Bp) {
-    }
+// TODO: POT AAQ support can be added by modifying these functions
+static InferenceEngine::Precision getInputPrecision() {
+    return InferenceEngine::Precision::I16;
+}
+static InferenceEngine::Precision getOutputPrecision() {
+    return InferenceEngine::Precision::I32;
+}
 
-    InferenceEngine::Precision getInputPrecision() const {
-        return _Ip;
+// TODO: These should take into account the HW platform limitations
+static InferenceEngine::Precision getWeightsPrecision(const LayerInfo& layer_info, const QuantizedLayerParams* quant_data) {
+    if (layer_info.isConvolution() || layer_info.isConvolutionFilter() || layer_info.isScaleShift()) {
+        return InferenceEngine::Precision::I16;
     }
-    InferenceEngine::Precision getWeightsPrecision() const {
-        return _Wp;
-    }
-    InferenceEngine::Precision getBiasesPrecision() const {
-        return _Bp;
-    }
-    InferenceEngine::Precision getOutputPrecision() const {
-        return _Op;
-    }
-};
 
-#define P_TYPE(X)\
-typename InferenceEngine::PrecisionTrait<InferenceEngine::Precision::X>::value_type
+    if (quant_data->_weights_quant.IsStatsSet()) {
+        // For networks with FakeQuantize layers
+        if (quant_data->_weights_quant.GetLevels() <= std::numeric_limits<uint8_t>::max()) {
+            return InferenceEngine::Precision::I8;
+        } else {
+            return InferenceEngine::Precision::I16;
+        }
+    } else {
+        return quant_data->_weights_low_precision ? InferenceEngine::Precision::I8 : InferenceEngine::Precision::I16;
+    }
+}
 
-#define PRECISION_TYPE(A, B, C, D, E)\
-    P_TYPE(A), P_TYPE(B), P_TYPE(C), P_TYPE(D), P_TYPE(E)
+// TODO: In future these should take into account the HW platform limitations
+static InferenceEngine::Precision getBiasesPrecision(const LayerInfo& layer_info, const QuantizedLayerParams* quant_data) {
+    if (layer_info.isConvolution() || layer_info.isConvolutionFilter() || layer_info.isScaleShift()) {
+        return InferenceEngine::Precision::I32;
+    }
 
+    if (quant_data->_weights_quant.IsStatsSet()) {
```

Comment:
```suggestion
    if (quant_data->_weights_quant.IsStatsSet()) {
        // For networks with FakeQuantize layers
        const auto fq_levels = quant_data->_weights_quant.GetLevels();
        if (fq_levels <= std::numeric_limits<uint8_t>::max()) {
            return InferenceEngine::Precision::fromType<gna_compound_bias_t>();
        }
        return InferenceEngine::Precision::I32;
    }
    if (quant_data->_inputs_low_precision) {
        return InferenceEngine::Precision::I8;
    }
    else if(getWeightsPrecision(layer_info, quant_data) == InferenceEngine::Precision::I8){
         return InferenceEngine::Precision::fromType<gna_compound_bias_t>();
    }
    return  InferenceEngine::Precision{InferenceEngine::Precision::I32};
      
```

Reply:
I'd leave it as it is for the same reason as stated earlier: one if-else pair divides this decision making function into two parts - one part for POT-processed models, the other for internal quantization. Soon the internal path might be removed.